src/main.rs

```diff
@@ -177,188 +225,225 @@ async fn sync(
                 break;
             }
+        };                        
+        for e in entries {                            
+
+            let blaze_version = get_version(e.resource.clone());
+            let blaze_version = match blaze_version {
+                Some(v) => v,
+                None => {                                    
+                    warn!("Could not read resource version from Blaze search set entry.");
+                    continue
+                }
+            };
+                                        
+            let resource_str = match serde_json::to_string(&e.resource) {
+                Ok(s) => s,
+                Err(_) => {
+                    "{}".to_string() 
+                }
+            };
+
+            match pg_versions.get(&blaze_version.resource_id) {
+                Some(pg_version) => { //Resource is already in pg
+                    if pg_version.version_id < blaze_version.version_id ||
+                        pg_version.version_id > blaze_version.version_id
+                    {
+                        //Resource exists in pg but is outdated. 
+                        //Add resource for batch update into pg
+                        update_counter += 1;                                        
+                        update_batch.push(PgUpdateItem {id: pg_version.pk_id, resource: resource_str.clone()});
+                        //Remove all entries from pg_versions that can be found in Blaze 
+                        //=> remaining ones need to be deleted from pg
+                        pg_versions.remove(&blaze_version.resource_id);  
+                    } else { //Blaze and pg versions match
+                        pg_versions.remove(&blaze_version.resource_id);
+                    }                                    
+                },
+                None => { // current resource not (yet) in pg
+                    //Add resource for batch insert into pg
+                        insert_counter += 1;                                    
+                        insert_batch.push(PgInsertItem {resource: resource_str.clone()});                                    
+                }
+            }
+
+            if update_counter > 0 && update_counter % batch_size == 0 {
+                update_helper(pg_con_pool, &update_batch, table_name).await?;
+                update_batch.clear();
+
+            } else if insert_counter > 0 && insert_counter % batch_size == 0 {
+                insert_helper(pg_con_pool, &insert_batch, table_name).await?;
+                insert_batch.clear();
+
+            }
+
+        }
+        // extract link to next page if exists or break
+        let next_url = search_set.link.iter()
+            .find(|link| link.relation == "next")
+            .map(|link| link.url.clone());
+
+        if let Some(next_url) = next_url {
+            url = next_url;
+        } else {
+            break;
         }
```

Comment:
Instead of a loop with this I would use a `while let` loop. Although you might need to fight a bit with the compiler maybe its not so simple.